module/execution/registers_async_test.go

```diff
@@ -0,0 +1,66 @@
+package execution
+
+import (
+	"testing"
+
+	"github.com/onflow/flow-go/model/flow"
+	state_synchronization "github.com/onflow/flow-go/module/state_synchronization/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitDataAvailable(t *testing.T) {
+	rootBlockHeight := uint64(1)
+	registersAsync := NewRegistersAsyncStore()
+	// test data available on init
+	registerID := unittest.RegisterIDFixture()
+	invalidRegisterID := flow.RegisterID{
+		Owner: "ha",
+		Key:   "ha",
+	}
+	registerValue1 := []byte("response1")
+	registerValue2 := []byte("response2")
+	require.False(t, registersAsync.initialized.Load())
+	firstHeight := rootBlockHeight
+	latestHeight := rootBlockHeight + 1
+
+	registers := storagemock.NewRegisterIndex(t)
+	registers.On("Get", registerID, firstHeight).Return(registerValue1, nil)
```

Comment:
similar to the comment on the above tests, it's quite difficult to read the tests when you stack up the setups of mocks and then relay on correct order of execution.

Reply:
ah so the basic idea behind the test format was:
```
test() {
  create common state
  
  setup common state behaviour
  
  test behaviour case 1
  test behaviour case 2
  ....
```
this was to reuse the code on the `create` and `setup` steps and mimic the real-life behavior of API calls on one instance at a certain state. 

I've switched it to separate steps for readability + parallelization